model/textparse/protobufparse.go

```diff
@@ -690,3 +744,40 @@ func isNativeHistogram(h *dto.Histogram) bool {
 		h.GetZeroThreshold() > 0 ||
 		h.GetZeroCount() > 0
 }
+
+func (p *ProtobufParser) convertToNHCB(t dto.MetricType) (*histogram.Histogram, *histogram.FloatHistogram, error) {
+	h := p.dec.GetHistogram()
+	temp := convertnhcb.NewTempHistogram()
+	v := h.GetSampleCountFloat()
+	if v == 0 {
+		v = float64(h.GetSampleCount())
+	}
+	if err := temp.SetCount(v); err != nil {
+		return nil, nil, err
+	}
+
+	if err := temp.SetSum(h.GetSampleSum()); err != nil {
+		return nil, nil, err
+	}
+	for _, b := range h.GetBucket() {
+		v := b.GetCumulativeCountFloat()
```

Comment:
Note to reviewers: the `convertnhcb` needs to be more smart and allow pre-determining if the histogram is integer or float. Now it figures it out itself, which is wasteful.

Reply:
added in fcb1b627775eda3d77cbbc93c1b3506f58870df9